arches_component_lab/views/api/widgets.py

```diff
@@ -1,51 +1,48 @@
 import json
 
+from django.forms.models import model_to_dict
 from django.utils import translation
 from django.views.generic import View
 
 from arches.app.models import models
 from arches.app.utils.response import JSONResponse
 
 
-class WidgetConfigurationView(View):
-    def get(self, request, graph_slug, node_alias):
-        user_language = translation.get_language()
+def update_i18n_properties(response):
+    user_language = translation.get_language()
+    config = json.loads(response["config"].value)
 
-        node = models.Node.objects.get(
-            graph__slug=graph_slug, alias=node_alias, source_identifier_id__isnull=True
-        )
+    if "i18n_properties" in config and isinstance(config["i18n_properties"], list):
+        for prop in config["i18n_properties"]:
+            if (
+                prop in config
+                and isinstance(config[prop], dict)
+                and user_language in config[prop]
+            ):
+                config[prop] = config[prop][user_language]
+    response["config"] = config
+    return response
 
+
+class WidgetConfigurationView(View):
+    def get(self, request, graph_slug, node_alias):
         card_x_node_x_widget = models.CardXNodeXWidget.objects.filter(
             node__graph__slug=graph_slug,
             node__alias=node_alias,
             node__source_identifier_id__isnull=True,
         ).first()
 
-        node_config = json.loads(node.config.value)
-        card_x_node_x_widget_config = json.loads(card_x_node_x_widget.config.value)
-
-        if node.datatype == "date":
-            # overwrites the card_x_node_x_widget_config with the node_config
-            response = {
-                **card_x_node_x_widget_config,
-                **node_config,
-            }
-        else:
-            # overwrites the node_config with the card_x_node_x_widget_config
-            response = {
-                **node_config,
-                **card_x_node_x_widget_config,
-            }
-
-        if "i18n_properties" in response and isinstance(
-            response["i18n_properties"], list
-        ):
-            for prop in response["i18n_properties"]:
-                if (
-                    prop in response
-                    and isinstance(response[prop], dict)
-                    and user_language in response[prop]
-                ):
-                    response[prop] = response[prop][user_language]
+        response = update_i18n_properties(model_to_dict(card_x_node_x_widget))
```

Comment:
I'd say use the JSONDeserializer.deserialize(JSONSerializer.serialize()) pattern instead, it was created for use with Arches models